include/picongpu/particles/atomicPhysics/AtomicPhysics.kernel

```diff
@@ -0,0 +1,362 @@
+                        printf("start kernel\n");
+                    }
+
+                    // get new histogram
+                    // Histogram histogram;
+                    PMACC_SMEM(
+                        acc, // mandatory first parameter
+                        histogram, // variable name
+                        Histogram // variable type
+                    );
+
+                    if(workerIdx == 0)
+                    {
+                        // cpu
+                        // std::cout << "    allocation histogram in SMEM" << std::endl;
+                        // gpu
+                        printf("    allocation histogram in SMEM\n");
+                    }
+
+                    if(workerIdx == 0)
+                    {
+                        // cpu
+                        // std::cout << "    allocated relative error instance" << std::endl;
+                        // gpu
+                        printf("    allocated relative error instance\n");
+                    }
+
+                    // init histogram and relative error
+                    if(workerIdx == 0)
+                    {
+                        // pass through parameters of histogram
+                        histogram.init(relativeErrorTarget, initialGridWidth);
+                    }
+                    cupla::__syncthreads(acc);
+                    // necessary since all access relativeError and histogram, but both are only valid after init
+
+                    // debug only
+                    if(workerIdx == 0)
+                    {
+                        // CPU
+                        // std::cout << "    init histogram" << std::endl;
+                        // gpu
+                        printf("    init histogram\n");
+                        // printf("        initialGridWidthSpecified: %f\n", initialGridWidth);
+                    }
+
+                    fillHistogram<T_numWorkers>(acc, electronBox, mapper, &histogram, atomicDataBox);
+                    cupla::__syncthreads(acc);
+
+                    // debug only, content of histogram, serial build only
+                    //{
+                    /*uint16_t numBins = histogram.getNumBins();
+                    float_X binLeftBoundary;
+                    float_X binCentralEnergy;
+
+                    std::cout << numBins << std::endl;
+
+                    std::cout << "{ " << std::endl;
+                    for(uint16_t i = 0u; i < numBins; i++)
+                    {
+                        std::cout << i << " : ";
+
+                        binLeftBoundary = histogram.getLeftBoundaryBin(i);
+                        binCentralEnergy = histogram.getEnergyBin(acc, i, atomicDataBox);
+
+                        std::cout << "L: " << binLeftBoundary << ", C: " << binCentralEnergy;
+                        std::cout << "|: " << histogram.getWeightBin(i) << std::endl;
+                    }
+                    std::cout << " }" << std::endl;
```

Comment:
please remove `std::cout` from the kernel.

Reply:
done